driver/utils/type_info.cpp

```diff
@@ -327,89 +327,93 @@ bool isStreamParam(SQLSMALLINT param_io_type) noexcept {
 }
 
 std::string convertCTypeToDataSourceType(const BoundTypeInfo & type_info) {
+    const auto set_nullability = [is_nullable = type_info.is_nullable] (const std::string & type_name) {
```

Comment:
That's rather an object than a function in this syntactic context.

Reply:
https://en.cppreference.com/w/cpp/language/lambda
Well, a `function object` is a grey area, but at the call site that looks like a function...

Reply:
Strictly speaking, its a closure, a code with data bound to it, i.e., same as (callable) instance of a Class, i.e., an object. :)
I think, it would look weird to have `smallCamelCase` named vars in function body. Frankly, I think, adopting anything but `snake_case` in C++ code, was a terrible idea, in general